safekeeper/src/receive_wal.rs

```diff
@@ -445,6 +448,12 @@ async fn network_write<IO: AsyncRead + AsyncWrite + Unpin>(
 /// walproposer, even when it's writing a steady stream of messages.
 const FLUSH_INTERVAL: Duration = Duration::from_secs(1);
 
+/// The metrics computation interval.
+///
+/// The Prometheus poll interval is 60 seconds at the time of writing. We sample the queue depth
+/// every 5 seconds, for 12 samples per poll. This will give a count of up to 12x active timelines.
+const METRICS_INTERVAL: Duration = Duration::from_secs(5);
```

Comment:
We could also just do a single sample per timeline per poll interval (every 60 seconds), such that the number of observations matches the number of timelines, but I think we probably want more frequent sampling.

Reply:
We still get only one sample ever 60s though, right? This interval places a bound on how stale the sample value can be. 5 seconds seems reasonable to me.

Reply:
Right, they're stored in mem